vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go

```diff
@@ -46,23 +46,30 @@ func NewProvider(calculator limitrange.LimitRangeCalculator,
 }
 
 // GetContainersResources returns the recommended resources for each container in the given pod in the same order they are specified in the pod.Spec.
+// If addAll is set to true, containers w/o a recommendation are also added to the list, otherwise they're skipped (default behaviour).
 func GetContainersResources(pod *core.Pod, vpaResourcePolicy *vpa_types.PodResourcePolicy, podRecommendation vpa_types.RecommendedPodResources, limitRange *core.LimitRangeItem,
-	annotations vpa_api_util.ContainerToAnnotationsMap) []vpa_api_util.ContainerResources {
+	addAll bool, annotations vpa_api_util.ContainerToAnnotationsMap) []vpa_api_util.ContainerResources {
 	resources := make([]vpa_api_util.ContainerResources, len(pod.Spec.Containers))
 	for i, container := range pod.Spec.Containers {
 		recommendation := vpa_api_util.GetRecommendationForContainer(container.Name, &podRecommendation)
 		if recommendation == nil {
-			klog.V(2).Infof("no matching recommendation found for container %s", container.Name)
-			continue
+			if addAll {
+				klog.V(2).Infof("no matching recommendation found for container %s, using Pod request", container.Name)
+				resources[i].Requests = container.Resources.Requests
+			} else {
+				klog.V(2).Infof("no matching recommendation found for container %s, skipping", container.Name)
+				continue
```

Comment:
We should either not continue at all or continue for both addAll=true and addAll=false (after the if addAll, else) and skip the else in line 63.

Reply:
The idea is that addAll controls whether we emit all containers, regardless of them having a recommendation or not.
If we always continue, there is no real change in behaviour.
Or did I misunderstand you comment?

Reply:
Ah sorry, I misread how big the else part is. 

Then we could do 
if !addAll {
    ...
    continue
}
so that you can drop the else

but it's not very important

Reply:
Changed, I trust your golang instincts way more than mine.